### DISCUSSION TITLE
Managing Sessions & Session Invalidating on its own.

### DISCUSSION THREAD

**Discussion by aarondyejr** [maxcountryman/axum-login#239] [Q&A] [answered]
Hey again, sorry for making so many discussions. I have a couple question I want to ask and something to inquire more about. 

1. What would be the best way to update a session (for example if someone changed information associated with said session)
2. Sessions are for some reason still invalidating themselves between restarts. I have monitored the issue, tried to track it down and I cannot seem to find where it is occurring at. I've thought maybe it was the store itself but I don't see any indications of that being the case, nor do I see any indications axum_login is at fault either. I am just lost on what to do. The cookie still exists, and the record still exists in the surrealdb database, yet axum_login returns 401 unauthorized when I access a protected route.
3. Is there anyway I can also accept passing in the cookie through an authorization header or would this require me making my own middleware to handle both cases?

**Comment by maxcountryman** [comment]
Can you explain what you mean by your first question: What specific situations are you concerned about?

For your second issue, please try to reproduce this problem using one of the examples within this repo. Once you have, please share it with us so we can take a look.

Finally, regarding your last question, please refer to this discussion: https://github.com/maxcountryman/axum-login/discussions/214

**Reply by aarondyejr** [reply]
Right so as far as my first question goes, I realize that no matter what they change (so long as its not their password) it the data updates as well when you fetch session information again.

As for my second issue, I was able to reproduce it using the sqlite example 1:1. It seems for some reason the cookie and session remain in the database and cookie storage, but when the server restarts it is no longer able to be found/recognized (I am also not using the memory store for sqlite, I used a .db file)

As for my third one, I greatly appreciate your response on that matter, I will look into it some more :)

**Reply by maxcountryman** [reply]
When using a memory store for sessions that message is expected after server restarts as I mentioned already.

If you still have that issue when not using a memory store for sessions then again, please share the code so that we can reproduce it.

**Reply by aarondyejr** [reply]
I wasn't using a memory store. I used your sqlite example but swapped it out for a file db instead of in memory. Unless I am mistaken on how sqlite works with sqlx, it makes no sense that it'd store it in memory and in the file as well.

**Reply by maxcountryman** [reply]
If you don't share the code there really isn't any way to help further.

**Reply by aarondyejr** [reply]
Here is how I instantiate the store and session: https://hastebin.skyra.pw/exofiwiqox.rust
This is how I handle impl Backend and AuthUser: https://hastebin.skyra.pw/azehoferun.rust

**Reply by maxcountryman** [reply]
Please fork this repo, modify one of the examples, and share the fork.

The goal is to help us reproduce what you think you're seeing, so we need to be able to run it directly ourselves.

**Reply by aarondyejr** [reply]
Here is the fork I made of the sqlite example that produces the same result as the one in my prod code. https://github.com/aarondyejr/axum-sqlite

**Reply by maxcountryman** [reply, marked as answer]
You are [regenerating a key](https://github.com/aarondyejr/axum-sqlite/blob/b80f091766f6680475075afdaa240e2ccf2de237/src/web/app.rs#L45) each time you start the server.

**Reply by aarondyejr** [reply]
Ah yknow what, that makes more sense looking back at it now. I should generate 1 single key and use it.

**Reply by aarondyejr** [reply]
Thank you so much max, sorry that seems like such a stupid issue ahaha.
